src/Compilers/Core/MSBuildTask/LocateEditorConfigFiles.cs

```diff
@@ -0,0 +1,161 @@
+        /// </summary>
+        [Required]
+        public ITaskItem[] InputFiles { get; set; }
+
+        public ITaskItem[] RootDirectories { get; set; }
+
+        [Output]
+        public ITaskItem[] EditorConfigFiles { get; private set; }
+
+        public LocateEditorConfigFiles()
+        {
+            TaskResources = ErrorString.ResourceManager;
+        }
+
+        public override bool Execute()
+        {
+            try
+            {
+                ExecuteCore();
+                return !Log.HasLoggedErrors;
+            }
+            catch (Exception ex)
+            {
+                // We'll explicitly log friendly errors and then rethrow in the case of IO errors while reading the file system.
+                // We'll only print the raw exception if we didn't already log something nice.
+                if (!Log.HasLoggedErrors)
+                {
+                    Log.LogErrorFromException(ex);
+                }
+
+                return false;
+            }
+        }
+
+        private void ExecuteCore()
+        {
+            var directoriesAddedToQueue = new HashSet<string>(StringComparer.Ordinal);
+            var directoriesToVisit = new Queue<DirectoryInfo>();
+            var editorConfigFiles = new List<ITaskItem>();
+
+            void addNewDirectoryIfNotAlreadyAdded(DirectoryInfo directory)
+            {
+                // If we're being passed files that don't exist, we'll just skip them and let the compiler handle the actual reporting.
+                if (directory.Exists)
+                {
+                    if (directoriesAddedToQueue.Add(directory.FullName))
+                    {
+                        directoriesToVisit.Enqueue(directory);
+                    }
+                }
+            }
+
+            foreach (var inputFile in InputFiles)
+            {
+                addNewDirectoryIfNotAlreadyAdded(new DirectoryInfo(Path.GetDirectoryName(Path.Combine(Path.GetFullPath(inputFile.ItemSpec)))));
+            }
+
+            while (directoriesToVisit.Count > 0)
+            {
+                var directory = directoriesToVisit.Dequeue();
+                var editorConfigFilePath = Path.Combine(directory.FullName, ".editorconfig");
+                bool isRootEditorConfig = false;
+
+                try
+                {
+                    if (File.Exists(editorConfigFilePath))
+                    {
+                        editorConfigFiles.Add(new TaskItem(editorConfigFilePath));
+
+                        // We will attempt to determine if this is a root .editorconfig, so we can stop enumerating farther if necessary.
+                        // We will do no further interpretation of the .editorconfig.
+
+                        isRootEditorConfig = FileIsRootEditorConfig(editorConfigFilePath);
+                    }
+                }
+                catch (IOException exception)
```

Comment:
We converted all file issues to IOException at the specific point when we did the operation. That way, other issues aren't being captured since that's probably a programming error.